sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/InterfaceMethodsShouldBeCallableByChildTypes.cs

```diff
@@ -0,0 +1,141 @@
+                {
+                    var classDeclaration = (ClassDeclarationSyntax)c.Node;
+                    var classSymbol = c.SemanticModel.GetDeclaredSymbol(classDeclaration);
+
+                    if (classSymbol == null ||
+                        classSymbol.IsSealed ||
+                        classSymbol.IsStatic ||
+                        !classSymbol.IsPublicApi() ||
+                        classDeclaration.Identifier.IsMissing)
+                    {
+                        return;
+                    }
+
+                    var allTrackedClassMembers = classSymbol.GetMembers().Where(IsTrackedMember);
+                    var explicitMembers = allTrackedClassMembers.Where(IsExplicit);
+                    var nonExplicitMembers = allTrackedClassMembers.Except(explicitMembers);
+
+                    explicitMembers.Where(member => !HasPublicEquivalentMember(member, nonExplicitMembers))
+                        .Select(faultyMember => CreateDiagnostic(classDeclaration, faultyMember))
+                        .WhereNotNull()
+                        .ToList()
+                        .ForEach(diagnostic => c.ReportDiagnostic(diagnostic));
+                }, SyntaxKind.ClassDeclaration);
+        }
+
+        private static bool IsTrackedMember(ISymbol symbol)
+        {
+            return symbol.Kind == SymbolKind.Event ||
+                symbol.Kind == SymbolKind.Property ||
+                (symbol.Kind == SymbolKind.Method && symbol.Name != ".ctor");
+        }
+
+        private static bool IsExplicit(ISymbol symbol)
+        {
+            var methodSymbol = symbol as IMethodSymbol;
+            if (methodSymbol != null)
+            {
+                return methodSymbol.ExplicitInterfaceImplementations.Length > 0;
+            }
+
+            var propertySymbol = symbol as IPropertySymbol;
+            if (propertySymbol != null)
+            {
+                return propertySymbol.ExplicitInterfaceImplementations.Length > 0;
+            }
+
+            var eventSymbol = symbol as IEventSymbol;
+            if (eventSymbol != null)
+            {
+                return eventSymbol.ExplicitInterfaceImplementations.Length > 0;
+            }
+
+            return false;
+        }
+
+        private static bool HasPublicEquivalentMember(ISymbol memberSymbolToMatch, IEnumerable<ISymbol> allMembers)
+        {
+            return allMembers.Any(
+                symbol =>
+                    symbol.Name == memberSymbolToMatch.Name &&
+                    symbol.Kind.Equals(memberSymbolToMatch.Kind) &&
+                    symbol.IsPublicApi());
+        }
+
+        private static Diagnostic CreateDiagnostic(ClassDeclarationSyntax classDeclaration, ISymbol faultyMember)
+        {
+            var syntaxNode = faultyMember.DeclaringSyntaxReferences.First().GetSyntax();
+
+            Location location;
+            switch (syntaxNode.Kind())
+            {
+                case SyntaxKind.MethodDeclaration:
```

Comment:
We could create a helper method here and reuse it in ParametersCorrectOrder.cs

Identifier GetIdentifier(SyntaxNode)